source/adapters/level_zero/v2/event.hpp

```diff
@@ -97,11 +106,16 @@ struct ur_event_handle_t_ : _ur_object {
   uint64_t getEventStartTimestmap() const;
   uint64_t getEventEndTimestamp();
 
+private:
+  ur_event_handle_t_(ur_context_handle_t hContext, event_variant hZeEvent,
+                     v2::event_flags_t flags, v2::event_pool *pool);
+
 protected:
   ur_context_handle_t hContext;
 
-  // non-owning handle to the L0 event
-  const ze_event_handle_t hZeEvent;
+  // Pool is used iff this is a pooled event
```

Comment:
```suggestion
  // Pool is used if this is a pooled event
```

Reply:
https://en.wikipedia.org/wiki/If_and_only_if

But probably worth typing it out in full.

Reply:
Didn't know that one :) Thanks.